packages/fetchai/skills/erc1155_deploy/handlers.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the handlers of the erc1155 deploy skill AEA."""
+
+from typing import Optional, cast
+
+from aea.configurations.base import ProtocolId
+from aea.crypto.base import LedgerApi
+from aea.decision_maker.messages.transaction import TransactionMessage
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.fipa.message import FIPAMessage
+from packages.fetchai.protocols.fipa.serialization import FIPASerializer
+from packages.fetchai.skills.erc1155_deploy.strategy import Strategy
+
+
+class FIPAHandler(Handler):
+    """This class implements a FIPA handler."""
+
+    SUPPORTED_PROTOCOL = FIPAMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        fipa_msg = cast(FIPAMessage, message)
+
+        if fipa_msg.performative == FIPAMessage.Performative.CFP:
+            self._handle_cfp(fipa_msg)
+        elif fipa_msg.performative == FIPAMessage.Performative.INFORM:
+            self._handle_inform(fipa_msg)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_cfp(self, msg: FIPAMessage) -> None:
+        """
+        Handle the CFP.
+
+        If the CFP matches the supplied services then send a PROPOSE, otherwise send a DECLINE.
+
+        :param msg: the message
+        :return: None
+        """
+        new_message_id = msg.message_id + 1
+        new_target = msg.message_id
+        self.context.logger.info(
+            "[{}]: received CFP from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        strategy = cast(Strategy, self.context.strategy)
+        contract = self.context.contracts.erc1155
+        self.context.logger.info(contract.item_ids)
```

Comment:
I really don't like these logger statements..no-one other than you will understand what this is once it prints to console.